src/data/set/basic.lean

```diff
@@ -2133,6 +2133,17 @@ end
 lemma univ_pi_eq_empty_iff : pi univ t = ∅ ↔ ∃ i, t i = ∅ :=
 by simp [← not_nonempty_iff_eq_empty, univ_pi_nonempty_iff]
 
+@[simp] lemma range_dcomp {β : ι → Type*} (f : Π i, α i → β i) :
+  range (λ (g : Π i, α i) i, f i (g i)) = pi univ (λ i, range (f i)) :=
```

Comment:
It always takes some time to me to parse this kind of statement. I feel like using
```
range (λ (g : Π i, α i), (λ i, f i (g i)))
```
would make things easier to parse.

Reply:
```suggestion
  range (λ (g : Π i, α i), (λ i, f i (g i))) = pi univ (λ i, range (f i)) :=
```